File: Frog jump with k distance/Recursion.py
def minimumJump(arr,size,k):
    if size == 0: return 0
    minimum = float('inf')

    for j in range(1,k+1):
        if size - j >= 0:
            minimum = min(minimum,abs(arr[size] - arr[size-j]) + minimumJump(arr,size-j,k))
        else: break 
    
    return minimum

def FrogJump():
    arr = [10, 5, 20, 0, 15]
    size = len(arr)-1
    k = 2

    return minimumJump(arr,size,k)

if __name__ == "__main__":
    print(FrogJump())

File: Frog jump with k distance/Recursion.java
public class Recursion {

    public static int FropJump(int size,int[] heights,int k) {
        if(size == 0) return 0;
        int minimumJump = Integer.MAX_VALUE;

        for(int j = 1;j <= k ;j ++) {
            if(size - j >= 0) {
                int jump = FropJump(size-j, heights, k) + Math.abs(heights[size] - heights[size-j]);
                minimumJump = Math.min(jump, minimumJump);
            }

            else {
                break;
            }
        }
        return minimumJump;
    }
    public static void main(String[] args) {
        int heights[] = {10, 5, 20, 0, 15};
        int size = heights.length;
        int k = 2;

        System.out.println(FropJump(size-1,heights,k));
    }
}